### ISSUE TITLE
Implemention and EMD Loss

### ISSUE THREAD

**Issue by zhulf0804** [vinits5/pcrnet_pytorch#10]
Hi vinits:
Thanks for your nice work and open source code. I find the following code is not good, because the normalization operation almost **removes the translation** between the source point cloud and the template point cloud. 

```
# mean substraction
source = source - torch.mean(source, dim=1, keepdim=True)
template = template - torch.mean(template, dim=1, keepdim=True)
```

I re-implemented the network in [https://github.com/zhulf0804/PCReg.PyTorch](https://github.com/zhulf0804/PCReg.PyTorch) with modifications(**iterative emd loss, partial registration, pairs point cloud generation, without mean substration, etc.**), I got the following visualization results:

![image](https://user-images.githubusercontent.com/36033208/101112223-cc059e00-3617-11eb-8b7a-88a6cb8bfbde.png)

I found EMD loss or CD loss is not friendly for point clouds registration with symmetrical structure.

**Comment by vinits5** [comment]
Hi @zhulf0804,
Thank you for your interest in our work. I understand your concern about the translation. The mean subtraction step is present in code as we follow the same settings as of ICP and PointNetLK algorithms. We have observed that even if you train the network without mean subtraction, the results will follow similar trends.
I understand that EMD and CD loss won't be very friendly in case of symmetrical structure or higher initial misalignments.

I see the partial-to-complete as a very interesting result. If you would be able to share the trained model with me then, I would be glad to have it on the repository for everyone's benefit. Of course, you will be acknowledged.

My contact details: vinitsarode5@gmail.com

**Comment by zhulf0804** [comment]
Hi vinits5,
Glad to hear from your. And i have sent the trained model by the email you provided above. 

I observed some partial-to-complete registration results may not good due to the simple network design, but most partial-to-complete registration results seems ok.

I am doing some attempts to optimize the model. Looking forward to communicating with you.

Best regards.
